Cargo.lock

```diff
@@ -5741,3 +5578,7 @@ checksum = "39f0c922f1a334134dc2f7a8b67dc5d25f0735263feec974345ff706bcf20b0d"
 dependencies = [
  "linked-hash-map",
 ]
+
+[[patch.unused]]
+name = "backtrace"
+version = "0.3.55"
```

Comment:
Since RLS has removed the backtrace dep, this is now coming from

https://github.com/rust-lang/rust/blob/dab3a80f235ed86d64d002e58b35adc3664e21a4/Cargo.toml#L109-L113

I wasn't sure if that entry should be removed from the root Cargo.toml given the presence of the submodule and a few other references (looked like the RA submodule has it, but feature gated) so left in for now and just removed it from the permitted dep lists in the tidy check to get rid of the error.

Reply:
I think it's fine to leave it for now, especially if RA still uses it conditionally (I assume we dist it without this feature enabled?).

cc @alexcrichton just in case (https://github.com/rust-lang/rust/commit/06d565c967bfb7c6ff52a991bbe47b4a2a25de3e)

Reply:
If nothing in rust-lang/rust is using the backtrace crate then this should be fine to remove. I thought Cargo at least would be using it through `anyhow`, but I was mistaken.

I think everything is using the dependency through the standard library now so it should be safe to remove.

Reply:
> I assume we dist it without this feature enabled?

Actually I guess it's completely disabled (at least at the moment)

https://github.com/rust-analyzer/rust-analyzer/blob/80ab753d7e0bf59b81df317d6ddda43cb919ec83/crates/stdx/Cargo.toml#L12-L19
```toml
[dependencies]
backtrace = { version = "0.3.44", optional = true }
always-assert = { version = "0.1.2", features = ["log"] }
# Think twice before adding anything here

[features]
# Uncomment to enable for the whole crate graph
# default = [ "backtrace" ]
```

> I think everything is using the dependency through the standard library now so it should be safe to remove.

SGTM :+1: if something changes in that RA crate or elsewhere which brings backtrace back then the patch can be restored accordingly

Reply:
Looks like bors will get to this before I get back to my dev machine. If so will open a followup PR to remove the patch